src/js/components/FileInput/FileInput.js

```diff
@@ -206,17 +206,14 @@ const FileInput = forwardRef(
         rightOffset = rightOffsetRemove + parseMetricToNum(rightPad);
       if (files.length === 1 || files.length > aggregateThreshold) {
         rightOffset =
-          rightOffsetBrowse +
-          rightOffsetRemove +
-          parseMetricToNum(theme.global.edgeSize.small) * 2;
+          rightOffsetBrowse + rightOffsetRemove + theme.fileInput.rightOffset;
       } else if (rightOffsetBrowse > rightOffsetRemove) {
-        rightOffset =
-          rightOffsetBrowse + parseMetricToNum(theme.global.edgeSize.small) * 2;
+        rightOffset = rightOffsetBrowse + theme.fileInput.rightOffset;
```

Comment:
maybe we can add to our discussion `rightOffset` not sure if I would know what that means.

the rest looks good though